File: Ruby/split_strings.rb
#Complete the solution so that it splits the string into pairs of two characters. If the string contains an odd number of characters then it should replace the missing second character of the final pair with an underscore ('_').

#Examples:

#* 'abc' =>  ['ab', 'c_']
#* 'abcdef' => ['ab', 'cd', 'ef']

def solution(str)
  # your code here
  str+="_" if str.length % 2 != 0
  result=[]
  (0...str.length).step(2) do |i|
    result<<str[i,2]
  end
  result
end

File: Python/split_strings.py
#Complete the solution so that it splits the string into pairs of two characters. If the string contains an odd number of characters then it should replace the missing second character of the final pair with an underscore ('_').

#Examples:

#* 'abc' =>  ['ab', 'c_']
#* 'abcdef' => ['ab', 'cd', 'ef']

def solution(s):
    
    if  len(s) %2 != 0:
      s += "_"

    result = []
    for i in range(0,len(s),2):
        result.append(s[i:i+2])
    print(result)
    return result
        
print(solution('abc'))